test/e2e/virtual/apiexport/virtualworkspace_test.go

```diff
@@ -229,7 +214,175 @@ func TestAPIExportVirtualWorkspace(t *testing.T) {
+	bindConsumerToProviderWithPermissionClaims(ctx, consumerWorkspace, serviceProviderWorkspace, t, kcpClients, identityHash)
+
+	t.Logf("create cowboy in consumer %v", consumerWorkspace)
+	createCowboyInConsumer(ctx, t, consumerWorkspace, wildwestClusterClient)
+	apifixtures.CreateSheriff(ctx, t, dynamicClients, consumerWorkspace, "wild.wild.west", "in-vw")
+
+	t.Logf("get virtual workspace client")
+	var apiExport *apisv1alpha1.APIExport
+
+	framework.Eventually(t, func() (bool, string) {
+		var err error
+		apiExport, err = kcpClients.Cluster(serviceProviderWorkspace).ApisV1alpha1().APIExports().Get(ctx, "today-cowboys", metav1.GetOptions{})
+		if err != nil {
+			return false, fmt.Sprintf("waiting on apiexport to be available %v", err.Error())
+		}
+		if len(apiExport.Status.VirtualWorkspaces) > 0 {
+			return true, ""
+		}
+		return false, "wating on virtual workspace to be ready"
+
+	}, wait.ForeverTestTimeout, 100*time.Millisecond, "waiting on virtual workspace to be ready")
+
+	apiExportVWCfg := rest.CopyConfig(cfg)
+	apiExportVWCfg.Host = apiExport.Status.VirtualWorkspaces[0].URL
+
+	wildwestVCClients, err := wildwestclientset.NewClusterForConfig(apiExportVWCfg)
+	require.NoError(t, err)
+
+	t.Logf("verify that the export resource is retrievable")
+	cowboysProjected, err := wildwestVCClients.Cluster(logicalcluster.Wildcard).WildwestV1alpha1().Cowboys("").List(ctx, metav1.ListOptions{})
+	require.NoError(t, err)
+	require.Equal(t, 1, len(cowboysProjected.Items))
+
+	t.Logf("Ensuring the appropriate core/v1 resources are available")
+	dynamicVWClients, err := dynamic.NewClusterForConfig(apiExportVWCfg)
+	require.NoError(t, err, "error creating dynamic cluster client for %q", apiExportVWCfg.Host)
+
+	grantedGVRs := []schema.GroupVersionResource{
+		{Version: "v1", Resource: "configmaps"},
+		{Version: "v1", Resource: "secrets"},
+		{Version: "v1", Resource: "serviceaccounts"},
+	}
+
+	for _, gvr := range grantedGVRs {
+		t.Logf("Trying to wildcard list %q", gvr)
+		_, err := dynamicVWClients.Cluster(logicalcluster.Wildcard).Resource(gvr).List(ctx, metav1.ListOptions{})
+		require.NoError(t, err, "error listing %q", gvr)
+	}
+
+	t.Logf("Verify that two sherrifs are eventually returned")
+	var ul *unstructured.UnstructuredList
+	framework.Eventually(t, func() (done bool, str string) {
+		gvr := schema.GroupVersionResource{Version: "v1", Resource: "sheriffs", Group: "wild.wild.west"}
+		var err error
+		ul, err = dynamicVWClients.Cluster(logicalcluster.Wildcard).Resource(gvr).List(ctx, metav1.ListOptions{})
+		if err != nil {
+			return false, err.Error()
+		}
+		if len(ul.Items) == 2 {
+			return true, "found two items"
+		}
+		return false, fmt.Sprintf("waiting on dynamic client to find both objects, found %v objects", len(ul.Items))
+	}, wait.ForeverTestTimeout, 100*time.Millisecond, "unable to wait for the two objects to be returned")
+
+	names := map[string]*struct{}{
```

Comment:
nit: you don't have to change this now but you could 

```go
require.Empty(t, cmp.Diff(
    []string{ul.Items[0].Name, ul.Items[1].Name},
    []string{"in-vw", "in-vw-before"},
))
```

Reply:
updated/.